ToDoApp/ToDoAppTests/ToDoServiceTests.swift

```diff
@@ -56,4 +61,21 @@ final class ToDoServiceTests: XCTestCase {
         }
         XCTAssertEqual(failure, RepositoryError.createOrUpdateError)
     }
+    
+    func testGetTask_Success() async {
+        let repositoryMock: RepositoryMock<TaskModel> = RepositoryMock(createOrUpdate: { value in
+            // Mock repository behaviour returns nil error
+            return RepositoryError.createOrUpdateError
+        }, read: {
+            return .success([TaskModel(id: UUID(), title: "Hi", deadline: Date(timeIntervalSince1970: 1672558922))])
+        })
+        let todoService = ToDoServiceImpl(dateChecker: Calendar.current, repo: repositoryMock)
+        let result = await todoService.getTasks()
+        guard case .success(let todoModel) = result else {
```

Comment:
As well the grouping logic should be tested. Repository returns a bunch of models, then it groups them in different sections depending on the deadline so, stubbed read method should return a bunch of models with different deadlines and we should verify that grouping was done correctly